StackExchange.Redis/StackExchange/Redis/GeoEntry.cs

```diff
@@ -0,0 +1,284 @@
+        /// <summary>
+        /// The possible options for the GeoRadius command
+        /// </summary>
+        public GeoRadiusOptions GeoRadiusOptions => geoRadiusOptions;
+        /// <summary>
+        /// The maximum number of results to return.
+        /// However note that internally the command needs to perform an effort proportional to the number of items matching the specified area, so to query very large areas with a very small COUNT option may be slow even if just a few results are returned.
+        /// </summary>
+        public int MaxReturnCount => maxReturnCount;
+        /// <summary>
+        /// Creates a new GeoRadius
+        /// </summary>
+        public GeoRadius(RedisKey key,GeoPosition geoPosition,double radius,int maxReturnCount =-1,GeoUnit geoUnit = GeoUnit.Meters,GeoRadiusOptions geoRadiusOptions = (GeoRadiusOptions.WITHCOORD | GeoRadiusOptions.WITHDIST | GeoRadiusOptions.WITHHASH))
+        {
+            this.key = key;
+            this.geoPosition = geoPosition;
+            this.radius = radius;
+            this.geoUnit = geoUnit;
+            this.geoRadiusOptions = geoRadiusOptions;
+            this.maxReturnCount = maxReturnCount;
+        }
+
+    }
+
+    /// <summary>
+    /// Describes the longitude and latitude of a GeoEntry
+    /// </summary>
+    public struct GeoPosition
+    {
+        internal readonly double longitude;
+        internal readonly double latitude;
+
+        /// <summary>
+        /// The Latitude of the GeoPosition
+        /// </summary>
+        public double Latitude => latitude;
+        /// <summary>
+        /// The Logitude of the GeoPosition
+        /// </summary>
+        public double Longitude => longitude;
+
+        /// <summary>
+        /// Creates a new GeoPosition
+        /// </summary>
+        /// <param name="longitude"></param>
+        /// <param name="latitude"></param>
+        public GeoPosition(double longitude,double latitude)
+        {
+            this.longitude = longitude;
+            this.latitude = latitude;
+        }
+
+        /// <summary>
+        /// See Object.ToString()
+        /// </summary>
+        public override string ToString()
+        {
+            return string.Format("{0} {1}", longitude, latitude);
+        }
+        /// <summary>
+        /// See Object.GetHashCode()
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return longitude.GetHashCode() ^ latitude.GetHashCode();
+        }
+        /// <summary>
+        /// Compares two values for equality
+        /// </summary>
+        public override bool Equals(object obj)
+        {
+            return obj is GeoPosition && Equals((GeoPosition)obj);
+        }
+        /// <summary>
+        /// Compares two values for equality
+        /// </summary>
+        public bool Equals(GeoPosition value)
+        {
+            return Equals(longitude, value.Longitude) && Equals(latitude, value.latitude);
```

Comment:
unless I'm missing something (reading in github, not the IDE), this is a boxing operation (4 boxes). Very inefficient. Should be just:

    return this.longitude == value.longitude && this.latitude == value.latitude;

Also; it may be desirable to only define the *implementation* of equality once; so `Equals` could also be written:

    return this == other;

(using the `==` operator below)